### ISSUE TITLE
Scenario Outlines skipped sometimes

### ISSUE THREAD

**Issue by KLBonn** [MicroFocus/bdd2octane#19]
We have issues with Scenario Outlines _sometimes_ not being processed correctly, e.g.:

> cannot find scenario: My Scenario Outline from feature
> cannot find scenario: My Scenario Outline from feature
> cannot find scenario: My Scenario Outline from feature
> ...
> 200 test case(s) processed
> 3 test case(s) skipped

Versions from our pom.xml:
```
        <java.version>17</java.version>
        <cucumber.version>7.15.0</cucumber.version>
        <junit.version>5.10.2</junit.version>
        <surefire.plugin.version>3.2.5</surefire.plugin.version>
```

The reason seems to be scenario outlines with matching prefix, i.e.

```
Feature: Outline breaker

Scenario Outline: My Scenario Outline
Then we do <action>

Examples:
  | action |
  | this    |
  | that    |

Scenario Outline: My Scenario Outline - now something different
Given we are prepared
Then we do <action>

Examples:
  | action |
  | this    |
  | that    |
```

Would you consider a PR for this?

**Comment by ghzhou** [comment]
@KLBonn , could you upload a sample JUnit XML report?

**Comment by KLBonn** [comment]
Actually I was able to build a modified version of bdd2octane fixing that issue. If you want, I could create a PR here containing my fix.

**Comment by bingyujin** [comment]
> I am not allowed to share the actual feature and report contents, but I created a small example closely matching the original data: [OutlineConfusion.zip](https://github.com/MicroFocus/bdd2octane/files/14648897/OutlineConfusion.zip)

Hi, do you provide the right cucumber.xml?

<testcase classname="Grundrisse mit aehnlichen Namen" name="Ein einfacher Grundriss - Example #1.1" time="0.048">
<testcase classname="Grundrisse mit aehnlichen Namen" name="Ein einfacher Grundriss - Example #1.2" time="0.002">
<testcase classname="Grundrisse mit aehnlichen Namen" name="Ein einfacher Grundriss mit einem Zusatz - Example #1.1">
<testcase classname="Grundrisse mit aehnlichen Namen" name="Ein einfacher Grundriss mit einem Zusatz 2 - Example #1.2">

For the 'name' attribute of the first 2 testcases, the left halves separated by "-" are the same; both of them are "Ein einfacher Grundriss".
However, in the second set of testcases, the left halves separated by "-" are different. One is "Ein einfacher Grundriss mit einem Zusatz", while the other one is "Ein einfacher Grundriss mit einem Zusatz 2", which contains the number 2.

**Comment by KLBonn** [comment]
You are right, I generated the cucumber.xml using my example project, and manually modified it to match the confidential one from my colleagues. The difference you spotted was my edit error. Please find attached a fixed version of the report.
[OutlineConfusion.zip](https://github.com/MicroFocus/bdd2octane/files/14692462/OutlineConfusion.zip)

If you require, I could provide the pom.xml, but then I would have to anonymize company-related attributes (and next week I am OOO).

**Comment by bingyujin** [comment]
> You are right, I generated the cucumber.xml using my example project, and manually modified it to match the confidential one from my colleagues. The difference you spotted was my edit error. Please find attached a fixed version of the report. [OutlineConfusion.zip](https://github.com/MicroFocus/bdd2octane/files/14692462/OutlineConfusion.zip)
> 
> If you require, I could provide the pom.xml, but then I would have to anonymize company-related attributes (and next week I am OOO).

Just obtaining the correct version of the XML is sufficient; there's no need for the pom.xml file. Thank you!

**Comment by bingyujin** [comment]
@KLBonn The issue was resolved on latest release 1.1.6
